### PR TITLE
Remove CentOS 6 from CI

### PR DESCRIPTION
**Pull request by bessbd**
## Overview

According to https://wiki.centos.org/FAQ/General#What_is_the_support_.27.27end_of_life.27.27_for_each_CentOS_release.3F  CentOS 6 is now discontinued. This PR removes it from the matrix.

## Checklist

- [ ] Code is written and works correctly
- [ ] Changes are covered by tests
- [ ] Any new configurable parameters are documented in `rel/overlay/etc/default.ini`
- [ ] A PR for documentation changes has been made in https://github.com/apache/couchdb-documentation

### CODE REVIEW

**Review by nickva — APPROVED** [review]

**Review by rnewson — APPROVED** [review]
+1 from me but would like @wohali or @janl to comment also before we merge.

**Comment by bessbd** [comment]
> +1 from me but would like @wohali or @janl to comment also before we merge.

I'll wait for @wohali or @janl 's +1 then before merging.

**Comment by janl** [comment]
I'm generally +1, but would also defer to @wohali for a final call :)

**Review by wohali — APPROVED** [review]
Sure, happy to drop for `main`.

Would be great if you could leave a comment in this commit as to what motivated this (the Unicode sort failure) and any research you have on it, because it will be lost if we leave it only on IRC.


We'll keep CentOS 6 packages for `3.x` for now, but when they start failing to build there, we'll drop it - best effort basis only.

**Comment by nickva** [comment]
@bessbd nice work discovering the bug, it was a rather tricky one!

**Comment by bessbd** [comment]
Motivation: as I was working on #3394 to get a green `main`, I was looking at the broken https://github.com/apache/couchdb/blob/main/src/mango/test/17-multi-type-value-test.py#L28 and I've noticed it's green everywhere but centos6.
I fired up the centos6 image in docker and realized couchdb `main` always returns one row less there than in my native (osx) environment.
After a long tracing and debugging session (with plenty of help from @nickva - thanks for that!), we've discovered that
`couch_ejson_compare:less([{[{<<"forename">>,<<"Eddie">>}]}], [{[{<<"ÿÿÿÿ">>,<<>>}]}]).` always returns `1` on centos6 and `-1` elswehere.
When I was trying to bump the libicu version, I've noticed the `yum` commands are failing so I looked up the supported centos versions and that's when I realized it's now discontinued - hence this PR.

@wohali : please let me know if I missed any important details here.

**Pull request merged**
